include/pm/pm.h

```diff
@@ -115,53 +115,15 @@ int pm_notifier_unregister(struct pm_notifier *notifier);
 const struct pm_state_info *pm_state_next_get(uint8_t cpu);
 
 /**
- * @}
- */
-
-/**
- * @brief System Power Management Constraints API
- * @defgroup subsys_pm_sys_constraint Constraints
- * @ingroup subsys_pm_sys
- * @{
- */
-
-/**
- * @brief Set a constraint for a power state
- *
- * @details Disabled state cannot be selected by the Zephyr power
- *	    management policies. Application defined policy should
- *	    use the @ref pm_constraint_get function to
- *	    check if given state is enabled and could be used.
- *
- * @note This API is refcount
- *
- * @param [in] state Power state to be disabled.
- */
-void pm_constraint_set(enum pm_state state);
-
-/**
- * @brief Release a constraint for a power state
- *
- * @details Enabled state can be selected by the Zephyr power
- *	    management policies. Application defined policy should
- *	    use the @ref pm_constraint_get function to
- *	    check if given state is enabled and could be used.
- *	    By default all power states are enabled.
- *
- * @note This API is refcount
- *
- * @param [in] state Power state to be enabled.
- */
-void pm_constraint_release(enum pm_state state);
-
-/**
- * @brief Check if particular power state is enabled
+ * @brief Inform the PM subsystem about a future event.
  *
- * This function returns true if given power state is enabled.
+ * This function can be used to inform the PM subsystem that an event is
+ * expected to happen in @p event_us microseconds. This information is used to
+ * influence the policy manager when deciding the next CPU power state.
  *
- * @param [in] state Power state.
+ * @param event_us Event time in microseconds.
  */
-bool pm_constraint_get(enum pm_state state);
+void pm_event_hint(uint32_t event_us);
```

Comment:
This is more than a hint IMHO. It is adding a timeout that will
trigger an interruption much likely waking up the SoC (or preventing
it from sleep). There is not much what the policy can do with it right
? Since the policy can distinguish this "hint" timeout from any other
scheduled event.

Reply:
This should influence the system because the next scheduled event may be the one provided by the hint, so, the policy will get that (in ticks) effectively influencing the next possible states. Do you have any other name suggestion?

Reply:
Yes, it will influence but more thant I hint IMHO. This will effectivelly trigger an event in the system (the way it is implemented). I thought it would just be used by the policy to influence which power state to select. Just affraid that it ends up causing an overhead triggerning many interruptions.

Reply:
@carlocaione can you review this?

Reply:
> it ends up causing an overhead triggerning many interruptions.

Why is that? you are literally informing the system that a new event will be likely to happen once in the future and to take that into account. I don't fully understand the concern for overhead here.

Reply:
My point is if should just be used by the policy to decide if it wants to wake up  earlier (or not even supspend) or not. Setting a timeout the policy does not know that this is a "hint". Basically you are enforcing an interruption to happen.
When I see a "hint" I interpret as an information that the policy can use to decide what is the best state to use but that is not case here. since the policy can't distinguish between this "hint" and a real event.